### PR TITLE
feat: Improve SDK for high-density board servers

### PR DESCRIPTION
**Pull request by muratozkan**

### CODE REVIEW

**Comment by muratozkan** [comment]
@syllabix @bhavaniMiro I'll incorporate changes from #14 into here

File: server/src/gameserver/sdk.rs [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    board_events: mpsc::Sender<BoardEvent>,
+    _health_checker: HealthChecker,
+}
+
+type BoardByUsers = HashMap<usize, HashSet<usize>>;
+
+impl Manager {
+    pub async fn setup() -> Result<Manager, Error> {
+        log::info!("connecting to agones sdk sidecar...");
+        let sdk = Sdk::new(None, None)
```
Comment (bhavaniMiro) [inline server/src/gameserver/sdk.rs:20]: Minor: Any reason `sdk` shouldn't be an attribute on `Manager` itself? then you won't have to keep passing `sdk` around.

**Review by bhavaniMiro — APPROVED** [review]

File: server/src/gameserver/sdk.rs (hunk 1 above)
Comment (syllabix) [reply server/src/gameserver/sdk.rs:20]: i think this could be a nice improvement - wdyt @muratozkan?

Comment (muratozkan) [reply server/src/gameserver/sdk.rs:20]: yeah, I'll try it out

**Comment by muratozkan** [comment]
No more conflicts :+1:

File: server/src/gameserver/sdk.rs (hunk 1 above)
Comment (muratozkan) [reply server/src/gameserver/sdk.rs:20]: @bhavaniMiro will merge this PR and do this separately so that everyone gets the fixes introduced here

**Pull request merged**